salt/matchers/list_match.py

```diff
@@ -3,16 +3,23 @@
 This is the default list matcher.
 '''
 from __future__ import absolute_import, print_function, unicode_literals
+import collections
+import six
 
 
 def match(tgt):
     '''
     Determines if this host is on the list
     '''
     try:
-        return __opts__['id'] == tgt \
-            or ',' + __opts__['id'] + ',' in tgt \
-            or tgt.startswith(__opts__['id'] + ',') \
-            or tgt.endswith(',' + __opts__['id'])
+        if isinstance(tgt, collections.Sequence) and not isinstance(tgt, six.string_types):
+            result = bool(__opts__['id'] in tgt)
```

Comment:
`__opts__['id'] in tgt` already returns a bool, so wrapping it in `bool()` is redundant.

Also, we can skip the type checking altogether by just relying on exception handling to narrow down what type the target is. I've opened https://github.com/cro/salt/pull/16 with one possible way we could do this.